packages/wdio-utils/src/shim.ts

```diff
@@ -289,8 +289,12 @@ export async function executeAsync(this: any, fn: Function, retries: Frameworks.
     this.wdioRetries = retries.attempts
 
     try {
+        /**
+         * Decrease the overall timeout by 3 milliseconds to ensure that when a test fails due to a timeout exception in Jasmine,
+         * the test objects are returned instead of being overridden by the values of the next test.
+         */
         // @ts-expect-error
-        const _timeout = this?._runnable?._timeout || globalThis.jasmine?.DEFAULT_TIMEOUT_INTERVAL || timeout
+        const _timeout = (this?._runnable?._timeout || globalThis.jasmine?.DEFAULT_TIMEOUT_INTERVAL || timeout) - 3
```

Comment:
Do we know that 3ms are enough for this to not happen? It seems a bit arbitrary.

Reply:
It does not need to be precisely 3ms. It just needs to be a shorter duration than the overall timeout. So, I've used 3ms, making sure it doesn't disrupt the timeouts set & execution timeline.

Reply:
Can we move the `3` to a const at the top of the file with a descriptive name and a comment as to why the value is chosen? I think it would greatly help anyone stumbling over it in the future.

Reply:
sure, on it.